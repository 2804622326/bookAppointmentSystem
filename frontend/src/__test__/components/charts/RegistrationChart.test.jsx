import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import RegistrationChart from '../../../components/charts/RegistrationChart.jsx';

jest.mock('recharts', () => ({
  ResponsiveContainer: ({ children }) => <div>{children}</div>,
  BarChart: ({ children }) => <div data-testid='bar-chart'>{children}</div>,
  XAxis: () => <div />,
  YAxis: () => <div />,
  Tooltip: () => <div />,
  Legend: () => <div />,
  Bar: () => <div />,
}));

jest.mock('../../../components/user/UserService.js', () => ({
  getAggregateUsersByMonthAndType: jest.fn(() => Promise.resolve({ data: { January: { VET: 2, PATIENT: 1 } } })),
}));

jest.mock('../../../components/common/NoDataAvailable.jsx', () => () => <div data-testid='no-data' />);

test('renders registration chart with data', async () => {
  const { asFragment } = render(<RegistrationChart />);
  await waitFor(() => screen.getByText(/users registration overview/i));
  expect(screen.queryByTestId('no-data')).not.toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});
