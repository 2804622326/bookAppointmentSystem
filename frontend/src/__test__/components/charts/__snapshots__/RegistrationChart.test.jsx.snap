// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders registration chart with data 1`] = `
<DocumentFragment>
  <section>
     
    <div>
      <h5
        class="chart-title mb-5"
      >
        Users Registration Overview
      </h5>
      <div
        data-testid="bar-chart"
      >
        <div />
        <div />
        <div />
        <div />
        <div />
        <div />
      </div>
    </div>
  </section>
</DocumentFragment>
`;
